lib/dry/system/auto_registrar.rb

```diff
@@ -32,7 +32,7 @@ def call(component_dir)
         components(component_dir).each do |component|
           next unless register_component?(component)
 
-          container.register(component.identifier, memoize: component.memoize?) { component.instance }
+          container.register(component.key, memoize: component.memoize?) { component.instance }
```

Comment:
Why not rely on duck-typing here? dry-container does `to_s` IIRC. This would be more future-proof.

Reply:
@solnic I considered it, yep, but I wondered whether `#to_s` actually made sense for the _whole component_ object as we pass it around within dry-system. The `Component` actually has a range of different responsibilities and it didn't feel right to reduce it down to its identifier only via `#to_s`.

Reply:
Happy to reconsider this and tweak in future, though!